engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/VectorTest.java

```diff
@@ -139,6 +143,90 @@ public void passingListOrArrayToEnsoAsArray() throws Exception {
     assertEquals("Hundred elements", 100, hundred.asInt());
   }
 
+  @Test
+  public void vectorWithWarningViaForEach() throws Exception {
+    warningsInContainer(1, 1);
+  }
+
+  @Test
+  public void arrayWithWarningViaForEach() throws Exception {
+    warningsInContainer(2, 1);
+  }
+
+  @Test
+  public void vectorToArrayToVectorWithWarningViaForEach() throws Exception {
+    warningsInContainer(3, 1);
+  }
+
+  @Test
+  public void insertSelfWithWarningViaForEach() throws Exception {
+    warningsInContainer(4, 1);
+  }
+
+  @Test
+  @Ignore // calling vector.slice drops warnings from element values
+  public void insertArgWithWarningViaForEach() throws Exception {
+    warningsInContainer(5, 1);
+  }
+
+  @Test
+  public void vectorWithWarningViaMap() throws Exception {
+    warningsInContainer(1, 2);
+  }
+
+  @Test
+  public void arrayWithWarningViaMap() throws Exception {
+    warningsInContainer(2, 2);
+  }
+
+  @Test
+  public void vectorToArrayToVectorWithWarningViaMap() throws Exception {
+    warningsInContainer(3, 2);
+  }
+
+  private void warningsInContainer(int type, int callType) throws Exception {
+    final URI srcUri = new URI("memory://warning.enso");
+    final Source src =
+        Source.newBuilder(
+                "enso",
+                """
+    from Standard.Base import Warning
+
+    cb f type call_type =
+      v = [Warning.attach "OKeyish" 20]
+      container = case type of
+        1 -> v
+        2 -> v.to_array
+        3 -> v.to_array.to_vector
+        4 -> (v+[42]).slice 0 1
+        5 -> ([42]+v).slice 1 2
+
+      case call_type of
+        1 -> container.each f
+        2 -> container.map f
+    """,
+                "warning.enso")
+            .uri(srcUri)
+            .buildLiteral();
+
+    var module = ctxRule.eval(src);
+    var cb = module.invokeMember("eval_expression", "cb");
```

Comment:
You don't have to define `srcUri`, `src`, neither `module`. It is enough to just call `var cb = ctxRule.getMethodFromModule("source code", "cb")`

Reply:
This is a common code pattern. Doesn't make sense to fix it at one place.